### PR TITLE
fix(core,platform): fix screen reader accessibility for reading mandatory field, includes all forms related components

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#4280

#### Please provide a brief summary of this pull request.
Screen reader is not able to read if any form field is mandatory.
providing fix for this in all form related components in core and platform.
components covered:
**core**: checkbox, combobox, date-picker, datetime-picker, file-uploader, input-group, multi-input, select, step-input, switch, time-picker
**platform**: checkbox, combobox, date-picker, datetime-picker, file-uploader, input, input-group, multi-input, multi-combobox, radio-group, select, step-input, slider, switch, text-area, time-picker

relevant example has been modified to show this behaviour; mostly form example for the component, if it is not there then first example in the component example page.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [NA] update `README.md`
- [NA] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by DeepakSap14** [comment]
@puru-hk , verify this once from your side if screen reader is able to read mandatory field. I have tested in Google chrome using "chromvox", it is working fine.

File: libs/platform/src/lib/components/form/checkbox-group/checkbox-group.component.html
Comment (kavya-b) [inline libs/platform/src/lib/components/form/checkbox-group/checkbox-group.component.html:1]: is this needed?

File: libs/platform/src/lib/components/form/input/input.component.ts
Comment (kavya-b) [inline libs/platform/src/lib/components/form/input/input.component.ts:62]: can we not have this in `base-input` file instead?

**Comment by puru-hk** [comment]
Checkbox Compoment, Radio butto group and textarea. Screen reader is reading but it is reding twice attached the screenshot

image

Date Picker: Title or heading is not reading 

Screenshot attached

image

Multi Input: Reading multiple time screen shot attached 

image

File: libs/platform/src/lib/components/form/checkbox-group/checkbox-group.component.html
Comment (DeepakSap14) [reply libs/platform/src/lib/components/form/checkbox-group/checkbox-group.component.html]: thank you, removed it.

File: libs/platform/src/lib/components/form/input/input.component.ts
Comment (DeepakSap14) [reply libs/platform/src/lib/components/form/input/input.component.ts:67]: required is input property in form-field component, from there while registering other form components, required field is set. But in case of Input component, as it is being used by other component, from parent component; required field need to be passed.

If we put required as input property in BaseInput, then it will be for all form-components. so field-component required input field will loose the mean.

**Comment by DeepakSap14** [comment]
> Checkbox Compoment, Radio butto group and textarea. Screen reader is reading but it is reding twice attached the screenshot
> 
> image
> 
> Date Picker: Title or heading is not reading
> 
> Screenshot attached
> 
> image
> 
> Multi Input: Reading multiple time screen shot attached
> 
> image

@puru-hk 
1. In this example, checkbox has label and title. both are being read.
2. Date picker label not being read is not in scope of this PR. this is happening because date picker has a different input property to set aria-label. this can be fixed with some other component. You can raise a different issue, if it not already raised.
3. I was not able to reproduce it. anyway, this will also be out of scope of this PR.

This PR is already touching more than 10 components, if we fix all other issues as well then it will be hard to review.

File: libs/platform/src/lib/components/form/input/input.component.ts
Comment (kavya-b) [reply libs/platform/src/lib/components/form/input/input.component.ts:67]: I see.. thanks for explaining.

File: libs/platform/src/lib/components/form/input/input.component.ts
Comment (kavya-b) [inline libs/platform/src/lib/components/form/input/input.component.ts:65]:
```suggestion
     * @hidden 
     * This `required` field is used internally by other components such as Input Group.
     * Input group will not be able to set required value to input, until `required` is input through the`@Input` property.
     */
```

**Review by kavya-b — APPROVED** [review]
checked for few components like checkbox, input and textarea and required field is read out by chromevox. LGTM, just one minor doc suggestion commented.

**Comment by puru-hk** [comment]
As discussed with @DeepakSap14  This has to be fixed for checkbox group component mandatory field is not reading as "Required".
Sceenshot attached 

image

**Comment by DeepakSap14** [comment]
> As discussed with @DeepakSap14 This has to be fixed for checkbox group component mandatory field is not reading as "Required".
> Sceenshot attached
> 
> image

Not able to reproduce on Macbook,chromevox and windows,JAWS 2021.

**Comment by puru-hk** [comment]
Tested In Jaws 2021 version it is working fine.

**Review by puru-hk — APPROVED** [review]

**Review by KevinOkamoto — APPROVED** [review]

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
We have a PR for form generation which is with high priority so for now blocking this until @N1XUS verifies it's ok to be merged. Thank for understanding.

**Comment by DeepakSap14** [comment]
form generation PR has been merged, this can be reviewed now.

File: libs/platform/src/lib/components/form/platform-file-uploader/platform-file-uploader.component.ts
Comment (N1XUS) [inline libs/platform/src/lib/components/form/platform-file-uploader/platform-file-uploader.component.ts:137]: Do we need it here?

Comment (DeepakSap14) [reply libs/platform/src/lib/components/form/platform-file-uploader/platform-file-uploader.component.ts:137]: @N1XUS 
it is required to connect form-field label with form-field input.
steps:
1. https://github.com/SAP/fundamental-ngx/blob/6ab69d630c1bbfc00e77cee1e708f1d3358c63a0/libs/platform/src/lib/components/form/base.input.ts#L196
2.https://github.com/SAP/fundamental-ngx/blob/6ab69d630c1bbfc00e77cee1e708f1d3358c63a0/libs/platform/src/lib/components/form/form-group/form-field/form-field.component.ts#L355

Comment (DeepakSap14) [reply libs/platform/src/lib/components/form/platform-file-uploader/platform-file-uploader.component.ts:137]: @N1XUS, please see my reply.
if all looks fine, please approve it.

**Review by N1XUS — APPROVED** [review]

**Pull request merged**
